deploy/olm-catalog/README.md

```diff
@@ -0,0 +1,72 @@
+# Operator Hub Release steps
```

Comment:
I think we should move these to a new file named `.github/RELEASE.md`. There are some additional release documentation that we can add in the future.

Reply:
Done @seanmalloy